com.unity.ml-agents.extensions/Runtime/Teams/BaseTeamManager.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections.Generic;
+using Unity.MLAgents;
+using Unity.MLAgents.Sensors;
+
+namespace Unity.MLAgents.Extensions.Teams
+{
+    public class BaseTeamManager : ITeamManager
+    {
+        readonly string m_Id = System.Guid.NewGuid().ToString();
```

Comment:
You could also make this an integer - have a global static ID and increment with `[Interlocked.Increment](https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.increment?view=net-5.0)` (which is threadsafe)